tests/lib/prepare-restore.sh

```diff
@@ -1,4 +1,12 @@
 #!/bin/bash
+set -x
+# NOTE: We must set -e so that any failures coming out of the various
+# statements we execute stops the build. The code is not (yet) written to
+# handle errors in general.
+set -e
```

Comment:
I personally would not include a comment why `set -e` is needed, in my world that is self-evident when dealing with shell.

Reply:
Well, it depends on the writing style. I think we should expand this comment to point out that the fact that we are executed rather than sourced is the key why this matters.

Reply:
I understand it depends, this is why I wrote: "in my world" ;) In any case, lets merge it.